### PR TITLE
SSCS-10934

### PR DESCRIPTION
**Pull request by mrmsali**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-10934

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by StevensonSi — APPROVED** [review]

**Review by AndreiTrushkov — APPROVED** [review]

**Review by amoslv1 — APPROVED** [review]

**Review by StevensonSi — DISMISSED** [review]

File: charts/sscs-tribunals-api/values.ccd.preview.template.yaml [hunk 1]
```diff
@@ -212,8 +212,11 @@ ccd:
     clusterName: "es"
     replicas: 1
   logstash:
-    image:
-      tag: ccd-sscs-logstash-latest
+    image:logstash:
```
Comment (ss-evoco) [inline charts/sscs-tribunals-api/values.ccd.preview.template.yaml:215]: Why did we change this ?

File: src/main/java/uk/gov/hmcts/reform/sscs/model/docassembly/WriteFinalDecisionTemplateContent.java [hunk 2]
```diff
@@ -161,13 +161,29 @@ public void addDescriptorTableIfPopulated(DescriptorTable desciptorTable) {
         }
     }
 
-    protected String getAppellantAndOtherPartyAttended(String hearingType, String appellantName, boolean presentingOfficerAttended, String bundlePage, String otherPartyNamesAttended) {
+    protected String getAppellantAndOtherPartyAttended(boolean appellantAttended, boolean appointeeAttended, String hearingType, String appellantName, String appointeeName, boolean presentingOfficerAttended, String bundlePage, String otherPartyNamesAttended) {
```
Comment (ss-evoco) [inline src/main/java/uk/gov/hmcts/reform/sscs/model/docassembly/WriteFinalDecisionTemplateContent.java:164]: This method could be made a lot more readable / viewer friendly.

File: src/main/java/uk/gov/hmcts/reform/sscs/model/docassembly/WriteFinalDecisionTemplateContent.java [hunk 3]
```diff
@@ -182,44 +198,70 @@ protected String getOtherPartyNamesAttendedString(String otherPartyNamesAttended
...
 
-    public List<String> getFaceToFaceTelephoneVideoHearingTypeSentences(String hearingType, String appellantName, String bundlePage,
-        boolean appellantAttended, boolean presentingOfficerAttended, String otherPartyNamesAttended) {
-        if (appellantAttended) {
-            if (equalsIgnoreCase("faceToFace", hearingType)) {
-                return singletonList("This has been an oral (face to face) hearing. "
-                    + getAppellantAndOtherPartyAttended(hearingType, appellantName, presentingOfficerAttended, bundlePage, otherPartyNamesAttended));
-            } else if (equalsIgnoreCase("triage", hearingType)) {
-                return singletonList(getTriageConsideredParagraph(bundlePage));
+    public List<String> getFaceToFaceTelephoneVideoHearingTypeSentences(String hearingType, String appellantName, String appointeeName, String bundlePage,
```
Comment (ss-evoco) [inline src/main/java/uk/gov/hmcts/reform/sscs/model/docassembly/WriteFinalDecisionTemplateContent.java:209]: This method is quite hard to follow. Can it be broken down ?

File: src/main/java/uk/gov/hmcts/reform/sscs/model/docassembly/WriteFinalDecisionTemplateContent.java [hunk 4]
```diff
@@ -182,44 +198,70 @@ protected String getOtherPartyNamesAttendedString(String otherPartyNamesAttended
...
+                            getConsideredParagraph(bundlePage, appellantName));
+                }
             }
         }
     }
 
     public void addHearingType(WriteFinalDecisionTemplateBody writeFinalDecisionTemplateBody) {
-        for (String hearingTypeSentence : getHearingTypeSentences(writeFinalDecisionTemplateBody.getAppellantName(), writeFinalDecisionTemplateBody.getPageNumber(),
-            writeFinalDecisionTemplateBody.getHearingType(), writeFinalDecisionTemplateBody.isAttendedHearing(), writeFinalDecisionTemplateBody.isPresentingOfficerAttended(),
+        for (String hearingTypeSentence : getHearingTypeSentences(writeFinalDecisionTemplateBody.getAppellantName(), writeFinalDecisionTemplateBody.getAppointeeName(), writeFinalDecisionTemplateBody.getPageNumber(),
```
Comment (ss-evoco) [inline src/main/java/uk/gov/hmcts/reform/sscs/model/docassembly/WriteFinalDecisionTemplateContent.java:263]: Might make more sense to just pass in the writeFinalDecisionTemplateBody

**Review by ss-evoco — CHANGES_REQUESTED** [review]

File: charts/sscs-tribunals-api/values.ccd.preview.template.yaml (hunk 1 above)
Comment (mrmsali) [reply charts/sscs-tribunals-api/values.ccd.preview.template.yaml:215]: I will change this back as I wasn't aware this had been changed.

File: src/main/java/uk/gov/hmcts/reform/sscs/model/docassembly/WriteFinalDecisionTemplateContent.java (hunk 4 above)
Comment (mrmsali) [reply src/main/java/uk/gov/hmcts/reform/sscs/model/docassembly/WriteFinalDecisionTemplateContent.java]: I have passed the writeFinalDecisionTemplateBody into the parameter.

File: src/main/java/uk/gov/hmcts/reform/sscs/model/docassembly/WriteFinalDecisionTemplateContent.java [hunk 5]
```diff
@@ -206,63 +267,208 @@ protected String getTriageConsideredParagraph(String bundlePage) {
...
+                        + notAttended;
+            }
+
+        }
+    }
+
+    protected String getNonFaceToFaceTypeSentences(boolean isAppointeeOnCase, boolean appointeeAttended,
+                                                   boolean appellantAttended, String hearingType,
+                                                   String appellantName, String appointeeName) {
+        String remoteHearing = "This has been a remote hearing in the form of a ";
```
Comment (ss-evoco) [inline src/main/java/uk/gov/hmcts/reform/sscs/model/docassembly/WriteFinalDecisionTemplateContent.java:437]: Aren't these three strings always used together? Why are they separate?

**Review by ss-evoco — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/model/docassembly/WriteFinalDecisionTemplateContent.java [hunk 6]
```diff
@@ -161,34 +174,82 @@ public void addDescriptorTableIfPopulated(DescriptorTable desciptorTable) {
...
+                        + getOtherPartyNamesAttendedString(otherPartyNamesAttended)
+                        + nonFaceToFaceHearing
+                        + bundlePage
+                        + ". "
                         + getPresentingOfficerAttendance(presentingOfficerAttended);
             }
         }
     }
 
+    protected String getFaceToFaceAttended(String attendedName, boolean appointeeAttended) {
```
[the remark this answers is not in the document]
Comment (mrmsali) [reply src/main/java/uk/gov/hmcts/reform/sscs/model/docassembly/WriteFinalDecisionTemplateContent.java:235]: I have removed one method and changed the method name to getAttended.

**Review by ss-evoco — DISMISSED** [review]

**Review by ss-evoco — DISMISSED** [review]

**Review by AndreiTrushkov — DISMISSED** [review]

**Review by AndreiTrushkov — APPROVED** [review]

**Review by AndreiTrushkov — DISMISSED** [review]

**Review by hajam09 — APPROVED** [review]
